pkg/reconciler/autoscaling/kpa/kpa_test.go

```diff
@@ -1060,6 +1062,35 @@ func TestReconcile(t *testing.T) {
 				withScales(0, -1), WithReachabilityReachable,
 				WithPAMetricsService(privateSvc), WithObservedGeneration(1),
 				WithPASKSNotReady(""),
+				func(pa *asv1a1.PodAutoscaler) {
+					pa.Status.ServiceName = testRevision
+				},
+			),
+		}},
+	}, {
+		Name: "initial scale zero: sks ServiceName empty",
+		Key:  key,
+		Ctx: context.WithValue(context.WithValue(context.Background(), asConfigKey{}, initialScaleZeroASConfig()), deciderKey{},
+			decider(testNamespace, testRevision, -1, /* desiredScale */
+				0 /* ebc */, scaling.MinActivators)),
+		Objects: []runtime.Object{
+			kpa(testNamespace, testRevision, withScales(0, -1), WithReachabilityReachable,
+				WithPAMetricsService(privateSvc), WithPASKSNotReady(noPrivateServiceName)),
+			// SKS won't be ready bc no ready endpoints, but private service name will be populated.
+			sks(testNamespace, testRevision, WithDeployRef(deployName), WithPrivateService, func(sks *nv1a1.ServerlessService) {
```

Comment:
Is the anoymous function even needed? It's empty by default anyway?

Reply:
Yeah it is empty by default. I thought it would make it clearer but it's probably not needed given the test case name.